[FVM] Implement program recovery for fungible tokens

turbolent opened the pull request:
Use the new program recovery feature of Cadence and add support for fungible tokens.

See https://github.com/onflow/flips/pull/283

Until the FLIP got approved, the feature is only enabled on all networks except Mainnet, for testing purposes.

SupunS commented on fvm/environment/program_recovery.go:
```diff
@@ -0,0 +1,203 @@
+const fungibleTokenVaultTypeIdentifier = "Vault"
+
+func isFungibleTokenContract(program *ast.Program, fungibleTokenAddress common.Address) bool {
+	const contractName = "FungibleToken"
```
a nitpick: maybe rename this variable to `fungibleTokenContract`/ `fungibleTokenContractName`, so the code below reads a bit better. e.g: `declaresConformanceTo(contractDeclaration, fungibleTokenContract)`

turbolent replied:
Good idea! Done in [6c46cee](https://github.com/onflow/flow-go/pull/6278/commits/6c46ceef0e63a9e5d6ddff8bf7bdd26cfa209298)

SupunS reviewed (approved):
LGTM!

turbolent commented:
Updated the PR: 
- The FVM feature is now gated and only enabled on all networks except Mainnet
- The replacement Fungible Token implementation is now updated to conform to the latest `FungibleToken` standard
- Added a test case for contract checking in the Cadence 1.0 migration
